truffleruby/src/main/java/org/truffleruby/core/thread/ThreadManager.java

```diff
@@ -391,4 +484,15 @@ public synchronized void accept(DynamicObject thread, Node currentNode) {
         }
     }
 
+    @TruffleBoundary
+    public void interrupt(Thread thread) {
+        final UnblockingAction action = unblockingActions.get(thread);
```

Comment:
I'm mildly worried that we never clean up the unblockingActions map. It's a big part of why I did it in the old PR. In an application churning through threads for IO this map will grow unbounded.